demos/groktunnel/main.go

```diff
@@ -100,22 +69,57 @@ func main() {
 	}
 }
 
+func serve(vmux *vhost.HTTPMuxer, host, port string) {
+	ml, err := vmux.Listen(net.JoinHostPort(host, port))
+	fatal(err)
+	srv := &http.Server{Handler: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		publicHost := strings.TrimSuffix(net.JoinHostPort(newSubdomain()+host, port), ":80")
+		pl, err := vmux.Listen(publicHost)
```

Comment:
Does this error out if this domain is already registered? Ideally you'd want it to fail if there's a collision with another active subdomain. When the original session ends you could de-register it to free up that domain.

Reply:
looks like go-vhost doesn't let you unregister, but could at least close the listener when finished, yea